### PR TITLE
Restoring from Borg backup will now do test list of repository

### PR DESCRIPTION
**Pull request by gozora**
##### Pull Request Details:

* Type: **Enhancement**

* Impact: **Normal**

* Reference to related issue (URL): https://github.com/rear/rear/issues/2029

* How was this pull request tested?
Full backup/restore on Fedora (NETFS and USB)

* Brief description of the changes in this pull request:
1. added checks whether we can read Borg archive, before any changes are written to disk
2. user can enable progress display (BORGBACKUP_SHOW_PROGRESS="yes") for `borg extract`
3. corrected mounting of USB device when using Borg back end

### CODE REVIEW

**Comment by gozora** [comment]
@jsmeix if you have a bit of spare time, please check whether you can spot something obviously wrong ...

Thanks!

V.

**Comment by jsmeix** [comment]
@gozora 
I was not in the office last Thusrday and Friday,
I have a look now...

**Comment by gozora** [comment]
@jsmeix 

> I was not in the office last Thusrday and Friday,
> I have a look now...

Thank you kind sir!

V.

File: usr/share/rear/verify/BORG/default/400_check_archive_access.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        (${choices[0]})
+            # Run 'less' with the original STDIN STDOUT and STDERR when 'rear' was launched by the user:
+            less $RUNTIME_LOGFILE 0<&6 1>&7 2>&8
+        ;;
+        (${choices[1]})
+            # rear_shell runs 'bash' with the original STDIN STDOUT and STDERR when 'rear' was launched by the user:
+            rear_shell "" "$rear_shell_history"
+        ;;
+        (${choices[2]})
+            # Continue recovery in migration mode:
```
Comment (jsmeix) [inline usr/share/rear/verify/BORG/default/400_check_archive_access.sh]: The `# Continue recovery in migration mode:` comment
is likely not right in this case here (probably via copy&paste)
and should be in this case here just something like
`# Continue recovery:`

Comment (gozora) [reply usr/share/rear/verify/BORG/default/400_check_archive_access.sh]: Right!  :thumbsup:

File: usr/share/rear/verify/BORG/default/400_check_archive_access.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Log "Command \"borg list $BORGBACKUP_OPT_REMOTE_PATH ${borg_dst_dev}${BORGBACKUP_REPO}\" returned: "
+borg_list
+
+rear_workflow="rear $WORKFLOW"
+choices[0]="View '$rear_workflow' log file ($RUNTIME_LOGFILE)"
+choices[1]="Use Relax-and-Recover shell and return back to here"
+choices[2]="Continue '$rear_workflow'"
+choices[3]="Abort '$rear_workflow'"
+
+prompt="Do you want to continue?"
```
Comment (jsmeix) [inline usr/share/rear/verify/BORG/default/400_check_archive_access.sh]: The `prompt` value is shown for each `while ... UserInput` loop
in contrast to the `LogPrint "Failed to list Borg archive."`
that scrolls away at the top.
Therefore the `prompt` value should show the reason of the
current `UserInput` dialog, for example something like
`prompt="Make Borg archive manually accessible"`

**Comment by jsmeix** [comment]
I did not spot something obviously wrong => Approved.

File: usr/share/rear/verify/BORG/default/400_check_archive_access.sh (hunk 1 above)
Comment (jsmeix) [reply usr/share/rear/verify/BORG/default/400_check_archive_access.sh]: Of course!
I know my code and see when you "steal" it ;-)
Of course you are welcome and I appreciate it when my code gets reused.

**Comment by gozora** [comment]
@jsmeix 

It is better described in [BORG/default/250_mount_usb.sh](https://github.com/rear/rear/blob/aa7b1979c347a3800959d757f7ce485fc10bd788/usr/share/rear/prep/BORG/default/250_mount_usb.sh)

```
# We need to mount USB_DEVICE in order to prepare for Borg archive
# initialization, but only if BORGBACKUP_HOST is not set.
# When BORGBACKUP_HOST is set, we don't need to mount anything as SSH
# backup destination will be handled internally by Borg it self.
```

so your guess

> I guess whether or not BORGBACKUP_HOST is empty is used
> to distinguish in the code between `Borg => SSH` versus `Borg => USB`
> as described in
> https://github.com/rear/rear/blob/master/doc/user-guide/04-scenarios.adoc
> but I am not a Borg user so that I may ask strange questions about Borg.

is right!

V.

File: usr/share/rear/verify/BORG/default/400_check_archive_access.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+borg_archive_cache_create && return
+
+LogPrint "Failed to list Borg archive."
+LogPrint "If you decide to continue, ReaR will partition your disks, but most probably will NOT be able to restore your data!"
+Log "Command \"borg list $BORGBACKUP_OPT_REMOTE_PATH ${borg_dst_dev}${BORGBACKUP_REPO}\" returned: "
+borg_list
+
+rear_workflow="rear $WORKFLOW"
+choices[0]="View '$rear_workflow' log file ($RUNTIME_LOGFILE)"
```
Comment (jsmeix) [inline usr/share/rear/verify/BORG/default/400_check_archive_access.sh]: I spotted something obviously wrong:
```
unset choices
```
is missing here, cf.
usr/share/rear/layout/prepare/default/300_map_disks.sh
wherefrom you copied it ;-)

**Comment by gozora** [comment]
Hello @jsmeix,

Your suggestions were applied by commit https://github.com/rear/rear/pull/2037/commits/085d4dc37d8ba3babda91cd6a0e1929bd0c7c834.

As always, many thanks for your inputs!

V.

**Pull request merged**